src/containers/terminal/PlaceOrder.js

```diff
@@ -205,13 +206,13 @@ class PlaceOrder extends React.Component {
     if(step.startsWith('1e')) {
       afterComma = parseInt(step.split('-')[1], 10);
     } else {
-      afterComma = (step.toString().split('.')[1] || '').length;
+      afterComma = (new BigNumber(step).toString(10).split('.')[1] || '').length;
     }
     const numberAfterComma = (string.split('.')[1] || '').length;
```

Comment:
Use bignumber to operate numbers inside of component. And call `toFixed(this.props.fixedLength)` only before rendering to string. Replace all mathematical operations with BigNumber methods calls:
* Mathematical: plus (add), minus (substract), times (multiply), div (divide).
* Logical: lt (less then), lte (.. or equal), gt (greater then), gte (... or equal), eq (equal).